sky/skylet/ray_patches/__init__.py

```diff
@@ -89,11 +86,21 @@ def patch() -> None:
     # by increasing the timeout.
     # Tracked in https://github.com/Azure/azure-cli/issues/20404#issuecomment-1249575110
     # Only patch it if azure cli is installed.
+    # Patch the azure_cli.py with `sed` instead of patching the file directly,
+    # because different versions of azure-cli-core have different line numbers.
     try:
         import azure
         from azure.identity._credentials import azure_cli
-        version = pkg_resources.get_distribution('azure-cli').version
-        _run_patch(azure_cli.__file__, _to_absolute('azure_cli.py.patch'),
-                   version)
+        version = pkg_resources.get_distribution("azure-cli-core").version
+        target_file = azure_cli.__file__
+        orig_file = f'{target_file}.{version}'
+        script = f"""\
+        if [ ! -f {orig_file} ]; then
+            echo Create backup file {orig_file}
+            cp {target_file} {orig_file}
+        fi
+        sed 's/kwargs["timeout"] = 10/kwargs["timeout"] = 30/g' {orig_file} > {target_file}
+        """
+        subprocess.run(script, shell=True, check=False)
```

Comment:
Switch to `sed` implementation, as the different version of the azure cli can have different line number causing the patching fail.

Reply:
Don't we pin the version of azure cli on remote clusters?